Mnemonic mdr

konraduh opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

teizenman commented:
@konraduh 
I noticed that the PR is in draft mode, if it's ready for review please press the `Ready for review` button below.

teizenman commented:
@konraduh 
Also - please merge from `demisto:master` branch into your branch.
You have many conflicts with the version that is already released.

konraduh commented:
Hi, yes, currently in draft mode: I need to figure out the merge conflicts first.

teizenman commented on Packs/mnemonicMDR/Integrations/ArgusManagedDefence/ArgusManagedDefence.py:
```diff
@@ -204,46 +255,270 @@ def test_module_command() -> str:
+        incidents.append(incident)
+
+    if result.get("data", []):
+        last_run["start_time"] = str(result.get("data")[-1]["createdTimestamp"] + 1)
```
Are you sure that this won't cause us to miss any incidents?

konraduh replied:
Yes. Otherwise I will get duplicates.

teizenman replied:
But wouldn't it cause you to miss incidents? What happens if there's an incident that needs to be fetched that its start time is just between the last `start_time` and the second you added?

konraduh replied:
Yes. We use epoch milliseconds. This logic has been tested. If we don't use `+1` we will get duplicates.

teizenman commented on Packs/mnemonicMDR/Integrations/ArgusManagedDefence/ArgusManagedDefence.yml:
```diff
@@ -1,84 +1,107 @@
-  display: First fetch time
+  required: true
+  options:
+  - Low
+  - Medium
+  - High
+  - Critical
```
Is there a reason to change it to uppercase?
It will break BC, and the code is still using the lowercase values.

konraduh replied:
It looks better in GUI. I believe if someone already has configured it to say e.g. "low", this would not affect this update as it runs lower() on it anyway. 
If, however, someone manage to give this new default value to a non-updated instance, it would break. But an old instance would still show lower-case values?

teizenman commented on Packs/mnemonicMDR/Integrations/ArgusManagedDefence/ArgusManagedDefence.yml:
```diff
@@ -164,29 +187,16 @@ script:
       required: true
-      secret: false
-    - default: false
+      description: 'Case ID '
```
```suggestion
      description: Case ID
```

konraduh replied:
Fixed with other changes.

teizenman commented on Packs/mnemonicMDR/Integrations/ArgusManagedDefence/ArgusManagedDefence.py:
```diff
@@ -451,17 +726,58 @@ def delete_comment_command(args: Dict[str, Any]) -> CommandResults:
 
-def download_attachment_command(args: Dict[str, Any]) -> Any:
+def download_attachment_by_filename_command(args: Dict[str, Any]) -> dict:
+    case_id = args.get("case_id", None)
+    file_name = args.get("file_name", None)
```
```suggestion
    case_id = args.get("case_id")
    file_name = args.get("file_name")
```

As `None` is the default default-value

teizenman commented on Packs/mnemonicMDR/Integrations/ArgusManagedDefence/ArgusManagedDefence.py:
```diff
@@ -570,6 +903,35 @@ def list_case_tags_command(args: Dict[str, Any]) -> CommandResults:
 
+def print_case_comments_command(args: Dict[str, Any]) -> List[Dict]:
+    case_id = args.get("case_id", None)
+    sort_by = args.get("sort_by", None)
```
This argument does not appear in the YML

teizenman commented on Packs/mnemonicMDR/pack_metadata.json:
```diff
@@ -2,8 +2,7 @@
     "name": "mnemonic MDR",
     "description": "Rapidly detect, analyse and respond to security threats with mnemonic’s leading Managed Detection and Response (MDR) service.",
     "support": "partner",
-    "certification": "certified",
```
Why did you remove it?

konraduh replied:
Did not know this pack was certified. :) I'll re-add.

teizenman reviewed (changes requested):
1. There are a lot of commented lines in the code and in the YML, please remove any unnecessary lines.
2. The commands in the readme are missing command examples and human readable outputs.
3. I didn't understand why you need the ignore `PB106` error.

teizenman commented on Packs/mnemonicMDR/pack_metadata.json:
```diff
@@ -36,4 +35,4 @@
         "MDR"
     ],
     "githubUser": []
```
Please update this field.

konraduh replied:
I thought this field was optional. Could you explain its function, please? I am a little uncertain about the meaning:

> (Optional) List of Github usernames to receive notification in the PR in case pack files were modified.

teizenman replied:
Whenever a file in this pack would change, the users that are configured inside the `githubUser` field will be notified.
It is highly encouraged for partners to fill this field in order to keep track of their packs.

konraduh commented on Packs/mnemonicMDR/Integrations/ArgusManagedDefence/ArgusManagedDefence.yml:
```diff
@@ -1,84 +1,107 @@
+  name: exclude_tag
+  type: 0
+  required: false
+  additionalinfo: 'Excludes fetching incidents with the optional tag. May be used
+    to exclude fetching Argus Cases created by XSOAR. Tags in Argus are of key: value
+    pairs. You may exclude with tag key, or key: value pairs by a comma-separated
+    string.'
```
Both are allowed. Added some clarification.

konraduh commented:
1. There are a lot of commented lines in the code and in the YML, please remove any unnecessary lines.
Fixed.
2. The commands in the readme are missing command examples and human readable outputs.
I discussed this with someone earlier (I don't remember who right now) during the initial phase of this contribution:  we deemed it a little unnecessary due to the sheer amount of commands that need documentation and relative little gain. There's one example command in the first part of the readme and all following commands follow the same pattern. I removed the sections from the readme. What do you think?
3. I didn't understand why you need the ignore PB106 error.
In the included playbook, I have set the commands to run only in the source instance it came from. If a user has two or more instances of the integration running in demisto where the instances have different access rights in Argus: every playbook command throws an unauthorised access exception, which is noisy as well as unnecessary traffic to our servers.

konraduh commented:
@teizenman is this anything I need to have a look at? I cannot seem to find where in the code this happens

> This pull request introduces 1 alert when merging 12489d2 into 3d18d84 - view on LGTM.com
>
> new alerts:
>
> 1 for Unnecessary pass

teizenman commented:
> 1. There are a lot of commented lines in the code and in the YML, please remove any unnecessary lines.
>    Fixed.

Great

> 2. The commands in the readme are missing command examples and human readable outputs.
>    I discussed this with someone earlier (I don't remember who right now) during the initial phase of this contribution:  we deemed it a little unnecessary due to the sheer amount of commands that need documentation and relative little gain. There's one example command in the first part of the readme and all following commands follow the same pattern. I removed the sections from the readme. What do you think?

I'm not sure who has told you this, but this is not following our standards, all of those fields should be filled in.

> 3. I didn't understand why you need the ignore PB106 error.
>    In the included playbook, I have set the commands to run only in the source instance it came from. If a user has two or more instances of the integration running in demisto where the instances have different access rights in Argus: every playbook command run in an instance which doesn't have access to the case throws an unauthorised access exception, which is noisy as well as unnecessary traffic to our servers.

Understood.

teizenman commented:
> @teizenman is this anything I need to have a look at? I cannot seem to find where in the code this happens
> 
> > This pull request introduces 1 alert when merging [12489d2](https://github.com/demisto/content/commit/12489d2e3bae0ae96f8cd71f1b9331815d54fb8d) into [3d18d84](https://github.com/demisto/content/commit/3d18d84926e57a6b804c585b214d27d848b6e460) - view on LGTM.com
> > new alerts:
> > 1 for Unnecessary pass

The alert is coming from the `.py` file, line 357, and it says: `Unnecessary 'pass' statement.`.
You can just delete the `pass` statement and it will be fixed.

konraduh commented:
I see now that we had command examples in the original release, but not command output. I have re-added command examples. I hope this is enough.

teizenman commented on Packs/mnemonicMDR/Integrations/ArgusManagedDefence/ArgusManagedDefence.py:
```diff
@@ -501,6 +500,10 @@ def append_demisto_entry_to_argus_case(case_id: int, entry: Dict[str, Any]) -> N
         file_name, file_extension = os.path.splitext(full_file_name)
         file_name = f"{file_name}{ATTACHMENT_SUBSTRING}{file_extension}"
         mime_type = mimetypes.guess_type(full_file_name)
+        if not mime_type[0]:
```
Is this access safe? Or `mime_type` can be `None`?

konraduh replied:
If mimetypes.guess_type() is not able to determine format, it will return a tuple None, None. I am going to improve this later. I found a bug on friday.

teizenman replied:
OK, please inform me when it's ready.

konraduh replied:
Could you please approve this merge, and the new functionality will come in a new PR.

teizenman commented:
> I see now that we had command examples in the original release, but not command output. I have re-added command examples. I hope this is enough.

Is there any problem providing the human readables as well? If not, please add them as well.

konraduh commented:
> > I see now that we had command examples in the original release, but not command output. I have re-added command examples. I hope this is enough.
> 
> Is there any problem providing the human readables as well? If not, please add them as well.

We would prefer not to. Otherwise we would manually just write "Test case, test description." The real value is context output.
